### DISCUSSION TITLE
Encountering Class 'Spatie\WebhookClient\Jobs\ProcessWebhookJob

### DISCUSSION THREAD

**Discussion by jennmagat** [spatie/laravel-webhook-client#133] [Q&A]
[2022-02-18 06:23:17] local.ERROR: Class 'Spatie\WebhookClient\Jobs\ProcessWebhookJob' not found {"exception":"[object] (Error(code: 0): Class 'Spatie\\WebhookClient\\Jobs\\ProcessWebhookJob' not found at /mnt/data/home/539903.cloudwaysapps.com/dnnzzpymsa/public_html/app/Handler/ProcessWebhook.php:10)
[stacktrace]
#0 /mnt/data/home/539903.cloudwaysapps.com/dnnzzpymsa/public_html/vendor/composer/ClassLoader.php(571): include()
#1 /mnt/data/home/539903.cloudwaysapps.com/dnnzzpymsa/public_html/vendor/composer/ClassLoader.php(428): Composer\\Autoload\\includeFile()
#2 [internal function]: Composer\\Autoload\\ClassLoader->loadClass()
#3 [internal function]: spl_autoload_call()
#4 /mnt/data/home/539903.cloudwaysapps.com/dnnzzpymsa/public_html/vendor/spatie/laravel-webhook-client/src/WebhookConfig.php(55): is_subclass_of()
#5 /mnt/data/home/539903.cloudwaysapps.com/dnnzzpymsa/public_html/vendor/spatie/laravel-webhook-client/src/WebhookClientServiceProvider.php(34): Spatie\\WebhookClient\\WebhookConfig->__construct()
#6 [internal function]: Spatie\\WebhookClient\\WebhookClientServiceProvider->Spatie\\WebhookClient\\{closure}()
#7 /mnt/data/home/539903.cloudwaysapps.com/dnnzzpymsa/public_html/vendor/laravel/framework/src/Illuminate/Collections/Collection.php(695): array_map()
#8 /mnt/data/home/539903.cloudwaysapps.com/dnnzzpymsa/public_html/vendor/spatie/laravel-webhook-client/src/WebhookClientServiceProvider.php(34): Illuminate\\Support\\Collection->map()
#9 /mnt/data/home/539903.cloudwaysapps.com/dnnzzpymsa/public_html/vendor/laravel/framework/src/Illuminate/Container/Container.php(873): Spatie\\WebhookClient\\WebhookClientServiceProvider->Spatie\\WebhookClient\\{closure}()
#10 /mnt/data/home/539903.cloudwaysapps.com/dnnzzpymsa/public_html/vendor/laravel/framework/src/Illuminate/Container/Container.php(758): Illuminate\\Container\\Container->build()
#11 /mnt/data/home/539903.cloudwaysapps.com/dnnzzpymsa/public_html/vendor/laravel/framework/src/Illuminate/Foundation/Application.php(851): Illuminate\\Container\\Container->resolve()
#12 /mnt/data/home/539903.cloudwaysapps.com/dnnzzpymsa/public_html/vendor/laravel/framework/src/Illuminate/Container/Container.php(694): Illuminate\\Foundation\\Application->resolve()
#13 /mnt/data/home/539903.cloudwaysapps.com/dnnzzpymsa/public_html/vendor/laravel/framework/src/Illuminate/Foundation/Application.php(836): Illuminate\\Container\\Container->make()
#14 /mnt/data/home/539903.cloudwaysapps.com/dnnzzpymsa/public_html/vendor/laravel/framework/src/Illuminate/Foundation/helpers.php(119): Illuminate\\Foundation\\Application->make()
#15 /mnt/data/home/539903.cloudwaysapps.com/dnnzzpymsa/public_html/vendor/spatie/laravel-webhook-client/src/WebhookClientServiceProvider.php(47): app()
#16 /mnt/data/home/539903.cloudwaysapps.com/dnnzzpymsa/public_html/vendor/laravel/framework/src/Illuminate/Container/Container.php(873): Spatie\\WebhookClient\\WebhookClientServiceProvider->Spatie\\WebhookClient\\{closure}()
#17 /mnt/data/home/539903.cloudwaysapps.com/dnnzzpymsa/public_html/vendor/laravel/framework/src/Illuminate/Container/Container.php(758): Illuminate\\Container\\Container->build()
#18 /mnt/data/home/539903.cloudwaysapps.com/dnnzzpymsa/public_html/vendor/laravel/framework/src/Illuminate/Foundation/Application.php(851): Illuminate\\Container\\Container->resolve()
#19 /mnt/data/home/539903.cloudwaysapps.com/dnnzzpymsa/public_html/vendor/laravel/framework/src/Illuminate/Container/Container.php(694): Illuminate\\Foundation\\Application->resolve()
#20 /mnt/data/home/539903.cloudwaysapps.com/dnnzzpymsa/public_html/vendor/laravel/framework/src/Illuminate/Foundation/Application.php(836): Illuminate\\Container\\Container->make()
#21 /mnt/data/home/539903.cloudwaysapps.com/dnnzzpymsa/public_html/vendor/laravel/framework/src/Illuminate/Routing/RouteDependencyResolverTrait.php(79): Illuminate\\Foundation\\Application->make()
#22 /mnt/data/home/539903.cloudwaysapps.com/dnnzzpymsa/public_html/vendor/laravel/framework/src/Illuminate/Routing/RouteDependencyResolverTrait.php(48): Illuminate\\Routing\\ControllerDispatcher->transformDependency()
#23 /mnt/data/home/539903.cloudwaysapps.com/dnnzzpymsa/public_html/vendor/laravel/framework/src/Illuminate/Routing/RouteDependencyResolverTrait.php(28): Illuminate\\Routing\\ControllerDispatcher->resolveMethodDependencies()
#24 /mnt/data/home/539903.cloudwaysapps.com/dnnzzpymsa/public_html/vendor/laravel/framework/src/Illuminate/Routing/ControllerDispatcher.php(41): Illuminate\\Routing\\ControllerDispatcher->resolveClassMethodDependencies()
#25 /mnt/data/home/539903.cloudwaysapps.com/dnnzzpymsa/public_html/vendor/laravel/framework/src/Illuminate/Routing/Route.php(262): Illuminate\\Routing\\ControllerDispatcher->dispatch()
#26 /mnt/data/home/539903.cloudwaysapps.com/dnnzzpymsa/public_html/vendor/laravel/framework/src/Illuminate/Routing/Route.php(205): Illuminate\\Routing\\Route->runController()
#27 /mnt/data/home/539903.cloudwaysapps.com/dnnzzpymsa/public_html/vendor/laravel/framework/src/Illuminate/Routing/Router.php(721): Illuminate\\Routing\\Route->run()
#28 /mnt/data/home/539903.cloudwaysapps.com/dnnzzpymsa/public_html/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(128): Illuminate\\Routing\\Router->Illuminate\\Routing\\{closure}()
#29 /mnt/data/home/539903.cloudwaysapps.com/dnnzzpymsa/public_html/vendor/laravel/framework/src/Illuminate/Routing/Middleware/SubstituteBindings.php(50): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
#30 /mnt/data/home/539903.cloudwaysapps.com/dnnzzpymsa/public_html/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(167): Illuminate\\Routing\\Middleware\\SubstituteBindings->handle()
#31 /mnt/data/home/539903.cloudwaysapps.com/dnnzzpymsa/public_html/vendor/laravel/framework/src/Illuminate/Routing/Middleware/ThrottleRequests.php(127): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
#32 /mnt/data/home/539903.cloudwaysapps.com/dnnzzpymsa/public_html/vendor/laravel/framework/src/Illuminate/Routing/Middleware/ThrottleRequests.php(103): Illuminate\\Routing\\Middleware\\ThrottleRequests->handleRequest()
#33 /mnt/data/home/539903.cloudwaysapps.com/dnnzzpymsa/public_html/vendor/laravel/framework/src/Illuminate/Routing/Middleware/ThrottleRequests.php(55): Illuminate\\Routing\\Middleware\\ThrottleRequests->handleRequestUsingNamedLimiter()
#34 /mnt/data/home/539903.cloudwaysapps.com/dnnzzpymsa/public_html/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(167): Illuminate\\Routing\\Middleware\\ThrottleRequests->handle()
#35 /mnt/data/home/539903.cloudwaysapps.com/dnnzzpymsa/public_html/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(103): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
#36 /mnt/data/home/539903.cloudwaysapps.com/dnnzzpymsa/public_html/vendor/laravel/framework/src/Illuminate/Routing/Router.php(723): Illuminate\\Pipeline\\Pipeline->then()
#37 /mnt/data/home/539903.cloudwaysapps.com/dnnzzpymsa/public_html/vendor/laravel/framework/src/Illuminate/Routing/Router.php(698): Illuminate\\Routing\\Router->runRouteWithinStack()
#38 /mnt/data/home/539903.cloudwaysapps.com/dnnzzpymsa/public_html/vendor/laravel/framework/src/Illuminate/Routing/Router.php(662): Illuminate\\Routing\\Router->runRoute()
#39 /mnt/data/home/539903.cloudwaysapps.com/dnnzzpymsa/public_html/vendor/laravel/framework/src/Illuminate/Routing/Router.php(651): Illuminate\\Routing\\Router->dispatchToRoute()
#40 /mnt/data/home/539903.cloudwaysapps.com/dnnzzpymsa/public_html/vendor/laravel/framework/src/Illuminate/Foundation/Http/Kernel.php(167): Illuminate\\Routing\\Router->dispatch()
#41 /mnt/data/home/539903.cloudwaysapps.com/dnnzzpymsa/public_html/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(128): Illuminate\\Foundation\\Http\\Kernel->Illuminate\\Foundation\\Http\\{closure}()
#42 /mnt/data/home/539903.cloudwaysapps.com/dnnzzpymsa/public_html/vendor/barryvdh/laravel-debugbar/src/Middleware/InjectDebugbar.php(67): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
#43 /mnt/data/home/539903.cloudwaysapps.com/dnnzzpymsa/public_html/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(167): Barryvdh\\Debugbar\\Middleware\\InjectDebugbar->handle()
#44 /mnt/data/home/539903.cloudwaysapps.com/dnnzzpymsa/public_html/packages/Webkul/Admin/src/Http/Middleware/Locale.php(32): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
#45 /mnt/data/home/539903.cloudwaysapps.com/dnnzzpymsa/public_html/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(167): Webkul\\Admin\\Http\\Middleware\\Locale->handle()
#46 /mnt/data/home/539903.cloudwaysapps.com/dnnzzpymsa/public_html/vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/TransformsRequest.php(21): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
#47 /mnt/data/home/539903.cloudwaysapps.com/dnnzzpymsa/public_html/vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/ConvertEmptyStringsToNull.php(31): Illuminate\\Foundation\\Http\\Middleware\\TransformsRequest->handle()
#48 /mnt/data/home/539903.cloudwaysapps.com/dnnzzpymsa/public_html/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(167): Illuminate\\Foundation\\Http\\Middleware\\ConvertEmptyStringsToNull->handle()
#49 /mnt/data/home/539903.cloudwaysapps.com/dnnzzpymsa/public_html/vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/TransformsRequest.php(21): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
#50 /mnt/data/home/539903.cloudwaysapps.com/dnnzzpymsa/public_html/vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/TrimStrings.php(40): Illuminate\\Foundation\\Http\\Middleware\\TransformsRequest->handle()
#51 /mnt/data/home/539903.cloudwaysapps.com/dnnzzpymsa/public_html/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(167): Illuminate\\Foundation\\Http\\Middleware\\TrimStrings->handle()
#52 /mnt/data/home/539903.cloudwaysapps.com/dnnzzpymsa/public_html/vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/ValidatePostSize.php(27): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
#53 /mnt/data/home/539903.cloudwaysapps.com/dnnzzpymsa/public_html/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(167): Illuminate\\Foundation\\Http\\Middleware\\ValidatePostSize->handle()
#54 /mnt/data/home/539903.cloudwaysapps.com/dnnzzpymsa/public_html/vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/PreventRequestsDuringMaintenance.php(86): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
#55 /mnt/data/home/539903.cloudwaysapps.com/dnnzzpymsa/public_html/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(167): Illuminate\\Foundation\\Http\\Middleware\\PreventRequestsDuringMaintenance->handle()
#56 /mnt/data/home/539903.cloudwaysapps.com/dnnzzpymsa/public_html/vendor/fruitcake/laravel-cors/src/HandleCors.php(38): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
#57 /mnt/data/home/539903.cloudwaysapps.com/dnnzzpymsa/public_html/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(167): Fruitcake\\Cors\\HandleCors->handle()
#58 /mnt/data/home/539903.cloudwaysapps.com/dnnzzpymsa/public_html/vendor/fideloper/proxy/src/TrustProxies.php(57): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
#59 /mnt/data/home/539903.cloudwaysapps.com/dnnzzpymsa/public_html/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(167): Fideloper\\Proxy\\TrustProxies->handle()
#60 /mnt/data/home/539903.cloudwaysapps.com/dnnzzpymsa/public_html/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(103): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
#61 /mnt/data/home/539903.cloudwaysapps.com/dnnzzpymsa/public_html/vendor/laravel/framework/src/Illuminate/Foundation/Http/Kernel.php(142): Illuminate\\Pipeline\\Pipeline->then()
#62 /mnt/data/home/539903.cloudwaysapps.com/dnnzzpymsa/public_html/vendor/laravel/framework/src/Illuminate/Foundation/Http/Kernel.php(111): Illuminate\\Foundation\\Http\\Kernel->sendRequestThroughRouter()
#63 /mnt/data/home/539903.cloudwaysapps.com/dnnzzpymsa/public_html/public/index.php(52): Illuminate\\Foundation\\Http\\Kernel->handle()
#64 /mnt/data/home/539903.cloudwaysapps.com/dnnzzpymsa/public_html/server.php(21): require_once('/mnt/data/home/...')
#65 {main}
"}

**Comment by jennmagat** [comment]
above issue is now fixed. I just downgraded 

  "spatie/laravel-webhook-client": "^2.75|^3.1",

including 2.75 fixed it. 

im just not sure now how to receive intercom webhooks from submit button.
right now my app only receiving pings from the intercom

**Comment by jennmagat** [comment]
Im encountering : The signature is invalid


[2022-02-20 12:34:34] local.ERROR: The signature is invalid. {"exception":"[object] (Spatie\\WebhookClient\\Exceptions\\WebhookFailed(code: 0): The signature is invalid. at /mnt/data/home/539903.cloudwaysapps.com/dnnzzpymsa/public_html/vendor/spatie/laravel-webhook-client/src/Exceptions/WebhookFailed.php:11)
[stacktrace]
#0 /mnt/data/home/539903.cloudwaysapps.com/dnnzzpymsa/public_html/vendor/spatie/laravel-webhook-client/src/WebhookProcessor.php(44): Spatie\\WebhookClient\\Exceptions\\WebhookFailed::invalidSignature()
#1 /mnt/data/home/539903.cloudwaysapps.com/dnnzzpymsa/public_html/vendor/spatie/laravel-webhook-client/src/WebhookProcessor.php(26): Spatie\\WebhookClient\\WebhookProcessor->ensureValidSignature()
#2 /mnt/data/home/539903.cloudwaysapps.com/dnnzzpymsa/public_html/vendor/spatie/laravel-webhook-client/src/WebhookController.php(11): Spatie\\WebhookClient\\WebhookProcessor->process()

**Comment by anandmainali** [comment]
i'm having same issue says "Spatie\WebhookClient\Jobs\ProcessWebhookJob" not found.

`Class "Spatie\WebhookClient\Jobs\ProcessWebhookJob" not found {"exception":"[object] (Error(code: 0): Class \"Spatie\\WebhookClient\\Jobs\\ProcessWebhookJob\" not found at /var/www/tac-backend/app/Handler/ProcessWebhook.php:12)
[stacktrace]
#0 /var/www/tac-backend/vendor/composer/ClassLoader.php(571): include()
#1 /var/www/tac-backend/vendor/composer/ClassLoader.php(428): Composer\\Autoload\\includeFile()
#2 [internal function]: Composer\\Autoload\\ClassLoader->loadClass()
#3 /var/www/tac-backend/vendor/laravel/framework/src/Illuminate/Queue/CallQueuedHandler.php(97): unserialize()
#4 /var/www/tac-backend/vendor/laravel/framework/src/Illuminate/Queue/CallQueuedHandler.php(253): Illuminate\\Queue\\CallQueuedHandler->getCommand()
#5 /var/www/tac-backend/vendor/laravel/framework/src/Illuminate/Queue/Jobs/Job.php(213): Illuminate\\Queue\\CallQueuedHandler->failed()
#6 /var/www/tac-backend/vendor/laravel/framework/src/Illuminate/Queue/Jobs/Job.php(192): Illuminate\\Queue\\Jobs\\Job->failed()
#7 /var/www/tac-backend/vendor/laravel/framework/src/Illuminate/Queue/Worker.php(581): Illuminate\\Queue\\Jobs\\Job->fail()
#8 /var/www/tac-backend/vendor/laravel/framework/src/Illuminate/Queue/Worker.php(527): Illuminate\\Queue\\Worker->failJob()
#9 /var/www/tac-backend/vendor/laravel/framework/src/Illuminate/Queue/Worker.php(455): Illuminate\\Queue\\Worker->markJobAsFailedIfWillExceedMaxAttempts()
#10 /var/www/tac-backend/vendor/laravel/framework/src/Illuminate/Queue/Worker.php(432): Illuminate\\Queue\\Worker->handleJobException()
#11 /var/www/tac-backend/vendor/laravel/framework/src/Illuminate/Queue/Worker.php(378): Illuminate\\Queue\\Worker->process()
#12 /var/www/tac-backend/vendor/laravel/framework/src/Illuminate/Queue/Worker.php(172): Illuminate\\Queue\\Worker->runJob()
#13 /var/www/tac-backend/vendor/laravel/framework/src/Illuminate/Queue/Console/WorkCommand.php(117): Illuminate\\Queue\\Worker->daemon()
#14 /var/www/tac-backend/vendor/laravel/framework/src/Illuminate/Queue/Console/WorkCommand.php(101): Illuminate\\Queue\\Console\\WorkCommand->runWorker()
#15 /var/www/tac-backend/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php(36): Illuminate\\Queue\\Console\\WorkCommand->handle()
#16 /var/www/tac-backend/vendor/laravel/framework/src/Illuminate/Container/Util.php(40): Illuminate\\Container\\BoundMethod::Illuminate\\Container\\{closure}()
#17 /var/www/tac-backend/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php(93): Illuminate\\Container\\Util::unwrapIfClosure()
#18 /var/www/tac-backend/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php(37): Illuminate\\Container\\BoundMethod::callBoundMethod()
#19 /var/www/tac-backend/vendor/laravel/framework/src/Illuminate/Container/Container.php(653): Illuminate\\Container\\BoundMethod::call()
#20 /var/www/tac-backend/vendor/laravel/framework/src/Illuminate/Console/Command.php(136): Illuminate\\Container\\Container->call()
#21 /var/www/tac-backend/vendor/symfony/console/Command/Command.php(298): Illuminate\\Console\\Command->execute()
#22 /var/www/tac-backend/vendor/laravel/framework/src/Illuminate/Console/Command.php(121): Symfony\\Component\\Console\\Command\\Command->run()
#23 /var/www/tac-backend/vendor/symfony/console/Application.php(1028): Illuminate\\Console\\Command->run()
#24 /var/www/tac-backend/vendor/symfony/console/Application.php(299): Symfony\\Component\\Console\\Application->doRunCommand()
#25 /var/www/tac-backend/vendor/symfony/console/Application.php(171): Symfony\\Component\\Console\\Application->doRun()
#26 /var/www/tac-backend/vendor/laravel/framework/src/Illuminate/Console/Application.php(94): Symfony\\Component\\Console\\Application->run()
#27 /var/www/tac-backend/vendor/laravel/framework/src/Illuminate/Foundation/Console/Kernel.php(129): Illuminate\\Console\\Application->run()
#28 /var/www/tac-backend/artisan(37): Illuminate\\Foundation\\Console\\Kernel->handle()
#29 {main}
"}`

**Reply by michaelklopf** [reply]
See the upgrade guide. For version 3 they moved the location of `ProcessWebhookJob`.
